### PR TITLE
Add: Astro to starter workflows

### PR DESCRIPTION
**Pull request by adrianmg**
## Pre-requisites

- [ ] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [ ] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [ ] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [ ] Should include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] Should specify least priviledge [permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) for `GITHUB_TOKEN` so that the workflow runs successfully.

**For _CI_ workflows, the workflow:**

- [ ] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [x] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [x] Packaging workflows should run on `release` with `types: [ created ]`.
- [ ] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [ ] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/code-scanning).
- [ ] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [ ] `name`: Name of the Code Scanning integration.
  - [ ] `creator`: Name of the organization/user producing the Code Scanning integration.
  - [ ] `description`: Short description of the Code Scanning integration.
  - [ ] `categories`: Array of languages supported by the Code Scanning integration.
  - [ ] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [ ] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [ ] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [ ] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [x] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

File: pages/astro.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    # - ".github/workflows/astro.yml"
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+env:
+  PAT: '.' # default value when not using subfolders
```
Comment (yoannchaudet) [inline pages/astro.yml:23]: The connotation with Personal Access Token is very strong here. I would either remove the option altogether (we don't have it for other Pages starter workflow) or find a better name.

Not too sure on a better name. `SITE_PATH`? `RELATIVE_PATH`? `BUILD_PATH`?

File: pages/astro.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Sample workflow for building and deploying an Astro site to GitHub Pages
+#
+# To get started with Astro see: https://docs.astro.build/en/getting-started/
+#
+
+name: Deploy Roadmap to Pages
```
Comment (yoannchaudet) [inline pages/astro.yml:6]:
```suggestion
name: Deploy Astro site to Pages
```

**Review by yoannchaudet — CHANGES_REQUESTED** [review]
We should include the default concurrency settings ([here](https://github.com/actions/starter-workflows/blob/ba77264174b7267793c7960a18718f941bf21b76/pages/nextjs.yml#L21-L24)).

And I'd like to wait a bit to see if we can get the setup for site/path entirely managed via the Astro CLI before merging in.

Context: https://github.com/actions/configure-pages/issues/30#issuecomment-1262598910

**Comment by oliverbenns** [comment]
@yoannchaudet 

I was about to commit a workflow but saw this in the PR template:

> Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.

But then I see a review for this PR, so now I'm a bit confused. Please advise.

File: pages/astro.yml (hunk 1 above)
Comment (adrianmg) [reply pages/astro.yml:23]: if we don't have it for other starters, let's remove it!

File: pages/astro.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Sample workflow for building and deploying an Astro site to GitHub Pages
```
Comment (Phantsure) [inline pages/astro.yml:1]: Do we not need an icon for this?

File: pages/astro.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+on:
+  push:
+    branches: [main]
+    # paths:
+    # only needed when project in subfolder
+    # - "subfolder"
+    # - ".github/workflows/astro.yml"
+  workflow_dispatch:
+
+permissions:
```
Comment (Phantsure) [inline pages/astro.yml:17]: permissions at workflow level can be `contents:read` and at job level add these permissions

**Comment by Phantsure** [comment]
@adrianmg Would you want to move it to draft if it is still being worked on?

**Comment by adrianmg** [comment]
I'm planning to work on this during this week

**Comment by adrianmg** [comment]
@yoannchaudet this is ready for review!

**Review by yoannchaudet — APPROVED** [review]
Looks good with me.

cc @actions/pages 

We should add Astro support in `actions/configure-pages` after.

**Comment by Phantsure** [comment]
Can we also have relevant properties and icon ?

**Comment by yoannchaudet** [comment]
@adrianmg Can you add a `astro.properties.json` just like: https://github.com/actions/starter-workflows/blob/main/pages/properties/nuxtjs.properties.json. This is where we keep metadata.

We will also want an icon (i.e. a svg) under: https://github.com/actions/starter-workflows/tree/main/icons

**Comment by adrianmg** [comment]
> @adrianmg Can you add a astro.properties.json just like: https://github.com/actions/starter-workflows/blob/main/pages/properties/nuxtjs.properties.json. This is where we keep metadata.
We will also want an icon (i.e. a svg) under: https://github.com/actions/starter-workflows/tree/main/icons

@yoannchaudet, done!

**Pull request merged**